be/src/olap/rowset/beta_rowset_writer.cpp

```diff
@@ -116,9 +118,11 @@ Status BetaRowsetWriter::add_block(const vectorized::Block* block) {
     return _add_block(block, &_segment_writer);
 }
 
-RowwiseIteratorUPtr BetaRowsetWriter::_get_segcompaction_reader(
-        SegCompactionCandidatesSharedPtr segments, std::shared_ptr<Schema> schema,
-        OlapReaderStatistics* stat, uint64_t* merged_row_stat) {
+std::unique_ptr<vectorized::VerticalBlockReader> BetaRowsetWriter::_get_segcompaction_reader(
```

Comment:
not return unique ptr, use std::unique_ptr<vectorized::VerticalBlockReader>* in input params